src/instructlab/config/init.py

```diff
@@ -183,6 +208,113 @@ def init(
     )
 
 
+def hw_auto_detect() -> tuple[str | None, int | None, _train, bool]:
+    # Third Party
+    import torch
+
+    gpus = 0
+    total_vram = 0
+    train = _train()
+    chosen_vram = None
+    chosen_profile_name = None
+    gpu_name = ""
+    edited_cfg = False
+    # try nvidia
+    if torch.cuda.is_available():
+        click.echo("Detecting Hardware...")
+        gpus = torch.cuda.device_count()
+        for i in range(gpus):
+            properties = torch.cuda.get_device_properties(i)
+            gpu_name = properties.name
+            total_vram += properties.total_memory  # memory in B
+    vram = int(floor(convert_bytes_to_proper_mag(total_vram)[0]))
+    if vram == 0:
+        return None, None, train, False
+    not_exact_match, chosen_vram, chosen_profile_name, train = lookup_card(
```

Comment:
This is confusing to read that we flip the value of matched inside of lookup_card when returned from the matching function.  Was there a good reason to flip the bool?

Reply:
good point, this is confusing. I don't feel strongly about the way I have it now so I'll switch it to make more sense